### DISCUSSION TITLE
Provision/MongoDBCluster

### DISCUSSION THREAD

**Discussion by dikshauat** [Sunbird-Ed/Community#169] [Sunbird ED configuration discussion]
Hi, Getting the following error on Provision/MongoDBCluster

TASK [mongodb-cluster : Install MongoDB package] *******************************
fatal: [10.128.0.23]: FAILED! => {
    "changed": false
}

MSG:

Failed to update apt cache: W:Updating from such a repository can't be done securely, and is therefore disabled by default., W:See apt-secure(8) manpage for repository creation and user configuration details., W:Target Packages (multiverse/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target Packages (multiverse/binary-all/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target Translations (multiverse/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target CNF (multiverse/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target CNF (multiverse/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, E:The repository 'http://www.apache.org/dist/cassandra/debian 39x Release' does not have a Release file.

PLAY RECAP *********************************************************************
10.128.0.23                : ok=6    changed=0    unreachable=0    failed=1    skipped=4    rescued=0    ignored=0   

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ rev
+ cut -d / -f1
+ git name-rev --name-only HEAD
+ rev
[Pipeline] sh
+ echo Provision/dev/Core/MongodbCluster
[Pipeline] echo
Could not find env specific Slack channel. Check for global slack channel..
[Pipeline] echo
Could not find slack environment variable. Skipping slack notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Provision/dev/Core/MongodbCluster
[Pipeline] echo
Could not find env specific email group. Check for global email group..
[Pipeline] echo
Could not find global email group variable. Skipping email notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE

**Comment by dikshauat** [comment]
Hi @keshavprasad, kindly look into the above issue and let me know it's solution.

Regards

**Reply by dikshauat** [reply]
Hi @keshavprasad, please help me with the issue.

Regards

**Reply by dikshauat** [reply]
Hi @Keshavprasad , @Shounk,

I removed the mongo_db_file_content key from the secret file and the roor is now gone. But now when I am running the job I am getting a different error :

TASK [mongodb-cluster : Install MongoDB package] *******************************
fatal: [10.128.0.23]: FAILED! => {
    "changed": false
}

MSG:

Failed to update apt cache: W:Updating from such a repository can't be done securely, and is therefore disabled by default., W:See apt-secure(8) manpage for repository creation and user configuration details., E:The repository 'http://old-releases.ubuntu.com/ubuntu bionic-security Release' does not have a Release file., W:Updating from such a repository can't be done securely, and is therefore disabled by default., W:See apt-secure(8) manpage for repository creation and user configuration details., W:Target Packages (multiverse/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target Packages (multiverse/binary-arm64/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target Packages (multiverse/binary-all/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target Translations (multiverse/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target CNF (multiverse/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target CNF (multiverse/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:1, W:Target Packages (multiverse/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:2, W:Target Packages (multiverse/binary-all/Packages) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:2, W:Target Translations (multiverse/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:2, W:Target CNF (multiverse/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:2, W:Target CNF (multiverse/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list.d/mongodb-org-4.4.list:1 and /etc/apt/sources.list.d/repo_mongodb_org_apt_ubuntu.list:2


PLAY RECAP *********************************************************************
10.128.0.23                : ok=6    changed=0    unreachable=0    failed=1    skipped=4    rescued=0    ignored=0   

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ git name-rev --name-only HEAD
+ rev
+ cut -d / -f1
+ rev
[Pipeline] sh
+ echo Provision/dev/Core/MongodbCluster
[Pipeline] echo
Could not find env specific Slack channel. Check for global slack channel..
[Pipeline] echo
Could not find slack environment variable. Skipping slack notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Provision/dev/Core/MongodbCluster
[Pipeline] echo
Could not find env specific email group. Check for global email group..
[Pipeline] echo
Could not find global email group variable. Skipping email notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE

Kindly look into it

**Reply by monojit18** [reply]
@keshavprasad: We are facing this issue while running the Provision -> MongoDb job and could not find any resolution so far.
Any suggestions?

**Reply by beepdot** [reply]
Please do not remove any variables mentioned. The error is probably due to missing mongo packages on your OS version. Can you please post the output of the below command
`lsb_release -a`
Based on the OS version, I can provide a few solutions.

**Comment by sharathkashyap** [comment]
@dikshauat I see that this thread has been inactive for quite some days now. Locking this thread from further conversation
